common/nixosModules/tailscale-backed-ec2.nix

```diff
@@ -0,0 +1,86 @@
+{ lib, config, pkgs, ... }:
+let
+  cfg = config.hacdc.tailscale;
+in
+{
+  imports = [
+    ./ec2.nix
+    ./ec2-impermanence.nix
+  ];
+
+  options.hacdc.tailscale = {
+    device-name = lib.mkOption {
+      type = lib.types.str;
+      description = "The factorio device name";
+      default = config.networking.hostName;
+    };
+    ssm-param = lib.mkOption {
+      type = lib.types.str;
+      description = "The SSM Paramstore key containing the Tailscale device key";
+    };
+    aws-role = lib.mkOption {
+      type = lib.types.str;
+      description = "The AWS IAM Role to assume in order to pull down the Tailscale key SSM parameter";
+    };
+  };
+
+  config = {
+    hacdc.impermanence.persistent-dirs = [
+      "/var/lib/tailscale"
+    ];
+
+    services.tailscale.enable = true;
+    systemd.services.tailscale-autoconnect = {
+      description = "Automatic connection to Tailscale";
+
+      # Make sure tailscale is running before trying to connect to tailscale
+      after = [
+        "network-pre.target"
+        "tailscaled.service"
+        "apply-ec2-data.service"
+      ];
+      wants = [
+        "network-pre.target"
+        "tailscaled.service"
+        "apply-ec2-data.service"
+      ];
+      wantedBy = [ "multi-user.target" ];
+
+      serviceConfig.Type = "oneshot";
+
+      path = with pkgs; [
+        tailscale
+        awscli
+        jq
+        curl
+      ];
+      script = ''
+        set -euo pipefail
+
+        hostname="${cfg.device-name}"
```

Comment:
For more robustness: https://teu5us.github.io/nix-lib.html#lib.strings.escapeshellarg

Reply:
Huh, TIL! Resolved in 99d65e65d53a2edb1c9cca6c1a171613c4f5d277